subsys/bluetooth/mesh/friend.c

```diff
@@ -1250,9 +1250,10 @@ static void subnet_evt(struct bt_mesh_subnet *sub, enum bt_mesh_key_evt evt)
 				BT_ERR("Failed updating friend cred for 0x%04x",
 				       frnd->lpn);
 				friend_clear(frnd);
-				break;
 			}
-
+			break;
```

Comment:
Should enqueue an update here as well though, right?

Reply:
I didn't see that the specification required to add the friend update. After receiving the net_key_update message. But adding this friend update does not affect the state of the LPN.

Reply:
You're right, the KR flag only changes in KR phase 2. I'll make a function for this rule in a new PR, this logic is repeated in various places. This works for this PR, though.